src/Stratis.Bitcoin/Base/Deployments/ThresholdConditionCache.cs

```diff
@@ -89,20 +89,34 @@ public List<ThresholdStateModel> GetThresholdStateMetrics(ChainedHeader indexPre
                 ChainedHeader periodStartsHeader = indexPrev.GetAncestor(indexPrev.Height - (currentHeight % period));
                 int periodEndsHeight = periodStartsHeader.Height + period;
 
+                var hexVersions = new Dictionary<string, int>();
+                int totalBlocks = 0;
+
                 while (indexPrev != periodStartsHeader)
                 {
                     if (this.Condition(indexPrev, deploymentIndex))
                     {
                         votes++;
                     }
 
+                    totalBlocks++;
+
+                    string hexVersion = indexPrev.Header.Version.ToString("X8");
```

Comment:
Its not correct to do this inside the loop of a single deployment if we had more then one deployment active we would have this data duplicated on each deployment result in the collection.

I would do a separate loop for the summery of block versions.

Reply:
Will keep it as is as per our discussion on slack.

Reply:
Good point you made on the confirmation window variable, I agree keep it as is.